server/lobbyconnection.py

```diff
@@ -505,14 +515,69 @@ async def check_policy_conformity(self, player_id, uid_hash, session, ignore_res
 
         return response.get("result", "") == "honest"
 
+    async def command_auth(self, message):
+        token = message["token"]
+        unique_id = message["unique_id"]
+        player_id = await self.oauth_service.get_player_id_from_token(token)
+
+        new_irc_password = hexlify(os.urandom(16)).decode()
+        await self.send({
+            "command": "irc_password",
+            "password": new_irc_password
+        })
+
+        async with self._db.acquire() as conn:
+            result = await conn.execute(
+                select([t_login.c.login, t_login.c.steamid])
+                .where(t_login.c.id == player_id)
+            )
+            row = await result.fetchone()
```

Comment:
Maybe we should have some handling for if the player id is not found. At least we should log a warning, and probably raise an authentication error with a useful message.